src/Lint.jl

```diff
@@ -345,31 +345,36 @@ function lintdir{T<:AbstractString}(dir::T, ctx::LintContext=LintContext())
     ctx.messages
 end
 
+function readandwritethestream(conn)
+    println("Connection accepted")
+    # Get file, code length and code
+    file = strip(readline(conn))
+    println("file: ", file)
+    code_len = parse(Int, strip(readline(conn)))
+    println("Code bytes: ", code_len)
+    code = Compat.UTF8String(read(conn, code_len))
+    println("Code received")
+    # Do the linting
+    msgs = lintfile(file, code)
+    # Write response to socket
+    for i in msgs
+        write(conn, string(i))
+        write(conn, "\n")
+    end
+    # Blank line to indicate end of messages
+    write(conn, "\n")
+    println("Connection closed")
+    close(conn)
+end
+
 function lintserver(port)
     server = listen(port)
     try
         println("Server running on port $port ...")
         while true
             conn = accept(server)
             @async try
-                println("Connection accepted")
-                # Get file, code length and code
-                file = strip(readline(conn))
-                println("file: ", file)
-                code_len = parse(Int, strip(readline(conn)))
-                println("Code bytes: ", code_len)
-                code = Compat.UTF8String(read(conn, code_len))
-                println("Code received")
-                # Do the linting
-                msgs = lintfile(file, code)
-                # Write response to socket
-                for i in msgs
-                    write(conn, string(i))
-                    write(conn, "\n")
-                end
-                # Blank line to indicate end of messages
-                write(conn, "\n")
-                println("Connection closed")
+              readandwritethestream(conn)
             catch err
               println("connection ended with error $err")
```

Comment:
as in review above, I think it's best to close the connection in a `finally` clause here.